irctest/server_tests/setname.py

```diff
@@ -0,0 +1,60 @@
+"""
+<https://ircv3.net/specs/extensions/setname>`_
+"""
+
+from irctest import cases
+from irctest.numerics import RPL_WHOISUSER
+
+
+class SetnameMessageTestCase(cases.BaseServerTestCase):
+    @cases.mark_specifications("IRCv3")
+    @cases.mark_capabilities("setname")
+    def testSetnameMessage(self):
+        self.connectClient("foo", capabilities=["setname"], skip_if_cap_nak=True)
+
+        self.sendLine(1, "SETNAME bar")
+        self.assertMessageMatch(
+            self.getMessage(1),
+            command="SETNAME",
+            params=["bar"],
+        )
+
+        self.sendLine(1, "WHOIS foo")
+        whoisuser = [m for m in self.getMessages(1) if m.command == RPL_WHOISUSER][0]
+        self.assertEqual(whoisuser.params[-1], "bar")
+
+    @cases.mark_specifications("IRCv3")
+    @cases.mark_capabilities("setname")
+    def testSetnameChannel(self):
+        """“[Servers] MUST send the server-to-client version of the
+        SETNAME message to all clients in common channels, as well as
+        to the client from which it originated, to confirm the change
+        has occurred.
+
+        The SETNAME message MUST NOT be sent to clients which do not
+        have the setname capability negotiated.“
+        """
+        self.connectClient("foo", capabilities=["setname"], skip_if_cap_nak=True)
+        self.connectClient("bar", capabilities=["setname"], skip_if_cap_nak=True)
+        self.connectClient("baz")
+
+        self.sendLine(1, "JOIN #chan")
```

Comment:
I think if you just use `self.joinChannel` here, instead of `sendLine` and then `getMessages`, that will remove the race condition

Reply:
Oh, actually it won't, you might still need one more `getMessages` call to clear the other client's JOIN lines